### PR TITLE
Starlark add-on

### PR DESCRIPTION
**Pull request by natnayr**
## Description

Add Starlark-based log transformation support to the AWS Shipper, enabling users to write custom parsing and transformation logic using Python-like Starlark scripts.

Fixes #CDS-2349

## Summary of Changes

- **New `StarlarkScript` parameter**: Configures custom log transformation with auto-detection of source type:
  - S3 bucket paths (`s3://bucket/key/transform.star`)
  - HTTP/HTTPS URLs
  - Base64-encoded scripts
  - Raw Starlark script code

- **Transformation capabilities**:
  - Unnest JSON arrays into individual log entries
  - Filter logs based on custom conditions
  - Transform log structure before sending to Coralogix
  - Enrich logs with additional metadata fields

- **Built-in helper functions**: `parse_json()`, `to_json()`, `log_debug()`

- **Template updates**: Added `StarlarkScript` parameter and S3 permissions for script fetching

## How Has This Been Tested?

- Unit tests for Starlark transformer compilation and execution
- Unit tests for JSON <-> Starlark value conversion
- Unit tests for script source detection (S3, URL, Base64, raw)
- Integration tests for S3 and URL script loading
- Property-based tests for transformation edge cases
- Fuzz testing for Starlark script compilation
- Benchmark tests for transformation performance
- Existing integration tests pass (`cargo test`)

# Checklist:
- [x] I have updated the versions in the SemanticVersion in template.yaml
- [x] I have updated the CHANGELOG.md
- [ ] I have created necessary PR to Terraform Module Repository (https://github.com/coralogix/terraform-coralogix-aws) if needed
- [ ] This change does not affect any particular component (e.g. it's CI or docs change)

### CODE REVIEW

**Comment by natnayr** [comment]
@codex review

**Comment by natnayr** [comment]
@codex review

**Comment by natnayr** [comment]
@codex review

**Comment by natnayr** [comment]
@codex review

File: README.md [hunk 1]
```diff
@@ -399,6 +400,113 @@ To enable the DLQ, provide the following parameters.
...
+
+### Built-in Functions
+
+The following helper functions are available in your Starlark scripts:
+
+| Function            | Description                                      |
+|---------------------|--------------------------------------------------|
+| `parse_json(str)`   | Parse a JSON string into a Starlark value        |
+| `to_json(value)`    | Convert a Starlark value to a JSON string        |
+| `log_debug(msg)`    | Log a debug message (visible when LogLevel=DEBUG)|
```
Comment (daidokoro) [inline README.md]:
```suggestion
| `print(msg)`    | Log a debug message (visible when LogLevel=DEBUG)|
```

File: src/logs/transform.rs [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+        }
+        return Ok(serde_json::Value::Object(map));
+    }
+
+    // Fallback: represent as string
+    Ok(serde_json::Value::String(value.to_string()))
+}
+
+/// Convert a Starlark result (expected to be a list) to JSON strings
+fn starlark_to_json_strings(value: Value) -> Result<Vec<String>, TransformError> {
```
Comment (daidokoro) [inline src/logs/transform.rs]: If the transform function returns a list, this function handles it, else, it wraps the result in an array and converts it to JSON. 

The issue is, what if an empty value is returned, for eg. `None` or an emptry string. Based on the logic, the empty value will be wrapped an array, something like `["null"]` and sent to the CX backend.

**Review by daidokoro — COMMENTED** [review]
Nice work @natnayr 
Made a few comments

**Comment by natnayr** [comment]
@codex review

**Comment by natnayr** [comment]
@codex review
